sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -163,27 +165,66 @@ class Iso8601TimestampFormatter(
   protected lazy val legacyFormatter = TimestampFormatter.getLegacyFormatter(
     pattern, zoneId, locale, legacyFormat)
 
+  override def parseOptional(s: String): Option[Long] = {
+    try {
+      val parsed = formatter.parseUnresolved(s, new ParsePosition(0))
+      if (parsed != null) {
+        val (epochSeconds, microsOfSecond) = extractSeconds(parsed)
+        Some(Math.addExact(Math.multiplyExact(epochSeconds, MICROS_PER_SECOND), microsOfSecond))
+      } else {
+        None
+      }
+    } catch {
+      case NonFatal(_) => None
```

Comment:
You're saying you're avoiding exceptions, but then this catches exceptions. Is this just the fall-back case for unexpected errors?

Reply:
Yes, we want to avoiding exceptions like `DateTimeParseException`, it cause by field value can't be parsed by format. There only catch JVM error like `VirtualMachineError`, same opeartion with `DateTimeUtils.stringToTimestamp`